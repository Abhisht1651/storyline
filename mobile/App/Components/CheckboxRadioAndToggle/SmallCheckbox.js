// @flow

import React, { useState } from 'react'
import { View, TouchableOpacity, Image } from 'react-native'
import Images from '../../Themes/Images'
import styles from '../Styles/CheckboxRadioAndToggleStyles/SmallCheckboxStyle'

type Props = {
  onPress: () => {}
}

const SmallCheckbox = (props: Props) => {
  const { onPress } = props
  const [checked, setChecked] = useState(false)

  const onPressHandler = () => {
    setChecked(!checked)
    if (!checked) onPress()
  }
  return (
    <TouchableOpacity style={styles.checkBox} activeOpacity={1} onPress={onPressHandler}>
      {checked && (
        <View style={styles.checkedBox}>
          <Image source={Images.checkmark} style={styles.checkmark} />
        </View>
      )}
    </TouchableOpacity>
  )
}

export default SmallCheckbox
