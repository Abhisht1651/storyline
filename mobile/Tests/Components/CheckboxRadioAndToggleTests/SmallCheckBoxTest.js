import { shallow } from 'enzyme'
import React from 'react'

// import { TouchableOpacity } from 'react-native'
import SmallCheckbox from '../../../App/Components/CheckboxRadioAndToggle/SmallCheckbox'

describe('SmallCheckbox', () => {
  it('should match to snapshot', () => {
    const component = shallow(<SmallCheckbox />)
    expect(component).toMatchSnapshot()
  })
})
