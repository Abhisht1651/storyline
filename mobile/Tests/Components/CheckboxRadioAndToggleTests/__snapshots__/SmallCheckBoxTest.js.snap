// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallCheckbox should match to snapshot 1`] = `
<ForwardRef
  activeOpacity={1}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#8F8FAF",
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 16,
      "justifyContent": "center",
      "width": 16,
    }
  }
/>
`;
